packages/driver/src/multi-domain/communicator.ts

```diff
@@ -144,15 +101,26 @@ export class SpecBridgeDomainCommunicator extends EventEmitter {
   private windowReference
 
   private handleSubjectAndErr = (data: any = {}, send: (data: any) => void) => {
-    const { subject, err, ...rest } = data
+    let { subject, err, ...rest } = data
+
+    // check to see if the 'err' key is defined, and if it is, we have an error of any type
+    const hasMultiDomainError = !!Object.getOwnPropertyDescriptor(data, 'err')
 
-    if (!subject && !err) {
+    if (!subject && !hasMultiDomainError) {
       return send(rest)
     }
 
     try {
+      if (hasMultiDomainError) {
+        try {
+          err = preprocessForSerialization(err)
+        } catch (e) {
+          err = e
+        }
+      }
+
       // We always want to make sure errors are posted, so clean it up to send.
-      send({ ...rest, subject, err: preprocessErrorForPostMessage(err) })
+      send({ ...rest, subject, err, hasMultiDomainError })
```

Comment:
Why is `hasMultiDomainError` necessary? I think we can just go off the existence of `err`, can't we?

Reply:
Unfortunately not quite. I'm going to take a stab at addressing this comment [here](https://github.com/cypress-io/cypress/pull/20520#discussion_r823792751)